adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -583,9 +655,22 @@ private boolean isBrokerWithPRTSupport(final Intent intent) {
      */
     public String getCurrentUser() {
         // authenticator is not used if there is not any user
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        if (accountList.length > 0) {
-            return accountList[0].name;
+        if (isBrokerAccountServiceSupported()) {
```

Comment:
Ooops, codeflow doesn't get my reply posted here. Sorry about that. 
I'm not sure i'm following. you should always opt-in to use broker and prompt user to grant the permission if it's missing (we'll send you an error back to let you know it's missing).